x-pack/plugins/lists/server/routes/list_item/patch_list_item_route.ts

```diff
@@ -32,7 +30,17 @@ export const patchListItemRoute = (router: ListsPluginRouter): void => {
       try {
         const { value, id, meta, _version } = request.body;
         const lists = await getListClient(context);
-        const listItem = await lists.updateListItem({
+
+        const dataStreamExists = await lists.getListItemDataStreamExists();
+        // needs to be migrated to data stream if index exists
+        if (!dataStreamExists) {
+          const indexExists = await lists.getListItemIndexExists();
+          if (indexExists) {
```

Comment:
same as one of the previous questions about possibility of `indexExists = false`

Reply:
similar to same questions, it's going to be handled below these lines

```
        const listItem = await lists.patchListItem({
          _version,
          id,
          meta,
          value,
        });
        if (listItem == null) {
          return siemResponse.error({
            body: `list item id: "${id}" not found`,
            statusCode: 404,
          });
        } else {
```